golang/internal/grpc/grpc.go

```diff
@@ -461,36 +462,36 @@ func executeVersionDeployRequest(
 	}
 }
 
-func streamContainerStatus(
+func streamContainerState(
```

Comment:
Both Kubernetes and Docker uses Status to describe workload progress, why the change?
In general terms I would say, state is something we will alter, status is something external to ourselves that we are aware of. I'm open for discussion.

Reply:
Docker definitely uses state:
```// ContainerState stores container's running state
// it's part of ContainerJSONBase and will return by "inspect" command
type ContainerState struct {
	Status     string // String representation of the container state. Can be one of "created", "running", "paused", "restarting", "removing", "exited", or "dead"
	Running    bool
	Paused     bool
	Restarting bool
	OOMKilled  bool
	Dead       bool
	Pid        int
	ExitCode   int
	Error      string
	StartedAt  string
	FinishedAt string
	Health     *Health `json:",omitempty"`
}```

Even though we convert that to a more simple structure,
```
export const CONTAINER_STATE_VALUES = ['running', 'waiting', 'exited', 'removed'] as const
export type ContainerState = (typeof CONTAINER_STATE_VALUES)[number]
```
we use State in the whole stack, so basically this one is the oddly named.

Reply:
Docker does it well too. They actually manage those states, they mutate those values. If you look deeper, in that State object, you can find 'statuses', because those are external to them, since they describe external processes. So they also follow this logic, as Kubernetes. We used it vice-versa, @robot9706 started correcting them not so long ago.

Reply:
So what is your proposal?

Reply:
Allrighty.